public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx

```diff
@@ -1,85 +1,63 @@
-import React, { PureComponent } from 'react';
+import React, { memo, useState, useEffect } from 'react';
 
 // Types
 import { ExploreQueryFieldProps } from '@grafana/data';
-import { FormLabel } from '@grafana/ui';
 
 import { PrometheusDatasource } from '../datasource';
 import { PromQuery, PromOptions } from '../types';
 
 import PromQueryField from './PromQueryField';
-export type Props = ExploreQueryFieldProps<PrometheusDatasource, PromQuery, PromOptions>;
-
-interface State {
-  interval: string;
-}
+import { PromExploreExtraField } from './PromExploreExtraField';
 
-export class PromExploreQueryEditor extends PureComponent<Props, State> {
-  // Query target to be modified and used for queries
-  query: PromQuery;
+export type Props = ExploreQueryFieldProps<PrometheusDatasource, PromQuery, PromOptions>;
 
-  constructor(props: Props) {
-    super(props);
-    const { query } = props;
-    this.query = query;
-    // Query target properties that are fully controlled inputs
-    this.state = {
-      // Fully controlled text inputs
-      interval: query.interval,
-    };
+export const PromExploreQueryEditor = memo(function PromExploreQueryEditor(props: Props) {
+  const { query, data, datasource, history, onChange, onRunQuery } = props;
+  const [step, setStep] = useState(query.interval);
+
+  function onChangeQueryStep(value: string, override?: boolean) {
+    const { query, onChange, onRunQuery } = props;
+    if (onChange) {
+      const nextQuery = { ...query, interval: value };
+      onChange(nextQuery);
+      if (override && onRunQuery) {
+        onRunQuery();
+      }
+    }
   }
 
-  onFieldChange = (query: PromQuery, override?: any) => {
-    this.query.expr = query.expr;
-  };
-
-  onIntervalChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
-    const interval = e.currentTarget.value;
-    this.query.interval = interval;
-    this.setState({ interval });
-  };
-
-  onRunQuery = () => {
```

Comment:
Seems like before we updated query upstream only when query was run. Now it seems to happen only in `onChangeQueryStep`. Is that intentional? Is that changing behaviour?

Reply:
It doesn't change behavior - we update query upstream using logic in `PromQueryField` when the query was run. There in `PromExploreQueryEditor` we care only about step value changes so we have a slightly simplified implementation there.